lib/Conversion/TritonGPUToLLVM/ConvertLayoutOpToLLVM.cpp

```diff
@@ -227,7 +229,16 @@ struct ConvertLayoutOpConversion
       elemTy = IntegerType::get(elemTy.getContext(), 64);
 
     auto llvmElemTy = getTypeConverter()->convertType(elemTy);
-
+    auto smemStrides = getStridesFromShapeAndOrder(
```

Comment:
smemStrides, smemOffsetVals, and SharedMemoryObject are the same for store/load and across reps, does it reduce registers if we move these outside the loop ?

Reply:
I guess it might be the same since LLVM can optimize it out.

Reply:
Yea that was my guess too

Reply:
Do you see any obvious opportunities to save on registers ? I think the main increase in register usage is `getSwizzledSharedPtrs`

Reply:
I'll think about it today